### PR TITLE
implement min/maxLength for string

### PR DESCRIPTION
**Pull request by mingwli**
#4567

### CODE REVIEW

File: src/docfx/lib/schema/JsonSchemaValidation.cs [hunk 1]
```diff
@@ -58,6 +58,24 @@ private static void ValidateScalar(JsonSchema schema, List<Error> errors, JValue
             {
                 errors.Add(Errors.UndefinedValue(JsonUtility.GetSourceInfo(scalar), scalar, schema.Enum));
             }
+
+            if ((schema.MaxLength.HasValue || schema.MinLength.HasValue) && CanTreatAsString())
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaValidation.cs]: Is checking the value good enough to detect whether it is a string?

```suggestion
            if ((schema.MaxLength.HasValue || schema.MinLength.HasValue) && scalar.Value is string str)
```

File: src/docfx/lib/schema/JsonSchemaValidation.cs [hunk 2]
```diff
@@ -58,6 +58,16 @@ private static void ValidateScalar(JsonSchema schema, List<Error> errors, JValue
             {
                 errors.Add(Errors.UndefinedValue(JsonUtility.GetSourceInfo(scalar), scalar, schema.Enum));
             }
+
+            if ((schema.MaxLength.HasValue || schema.MinLength.HasValue) && scalar.Value is string str)
```
Comment (OsmondJiang) [inline src/docfx/lib/schema/JsonSchemaValidation.cs]: do we need throw exception here if value is not string?

Comment (mingwli) [reply src/docfx/lib/schema/JsonSchemaValidation.cs]: tried that at "https://www.jsonschemavalidator.net/", no validation error thrown.

File: src/docfx/Errors.cs [hunk 3]
```diff
@@ -364,9 +364,18 @@ public static Error UnexpectedType(SourceInfo source, string expectedType, strin
...
             => new Error(ErrorLevel.Warning, "undefined-value", $"Value '{value}' is not accepted. Valid values: {Join(validValues)}", source);
 
+        /// <summary>
+        /// Array length not within min and max.
+        /// </summary>
         public static Error ArrayLengthInvalid(SourceInfo source, string propName, int? minItems = null, int? maxItems = null)
             => new Error(ErrorLevel.Warning, "array-length-invalid", $"Array {(string.IsNullOrEmpty(propName) ? "" : $"'{propName}' ")}length should be {(minItems.HasValue ? $">= {minItems.Value}" : $"<= {maxItems.Value}")}", source);
 
+        /// <summary>
+        /// String length not with min and max.
```
Comment (live1206) [inline src/docfx/Errors.cs]:
```suggestion
        /// String length not within min and max.
```

**Pull request merged**